src/psyclone/tests/domain/lfric/lfric_extract_driver_creator_test.py

```diff
@@ -0,0 +1,510 @@
+                          ("real_x", "15.28.2_real_X_builtin.f90")
+                          ])
+def test_lfric_driver_unsupported_builtins(name, filename, capsys):
+    '''The following builtins do not have a proper lower_to_language_level
+    method to create the PSyIR, so they are not supported in the driver
+    creation: LFRicXInnerproductXKern, LFRicSumXKern, LFRicIntXKern,
+    LFRicRealXKern. This tests also the error handling of the functions
+    write_driver, get_driver_as_string, create. '''
+
+    _, invoke = get_invoke(filename, API, dist_mem=False, idx=0)
+
+    driver_creator = LFRicExtractDriverCreator()
+
+    # The create method should raise an exception
+    # -------------------------------------------
+    with pytest.raises(NotImplementedError) as err:
+        # The parameters do not really matter
+        driver_creator.create(invoke.schedule, [], [], "extract", "_post",
+                              region_name=("region", "name"))
+    assert f"LFRic builtin '{name}' is not supported" in str(err.value)
+
+    # The get_driver_as_string method returns
+    # an empty string, but prints an error message
+    # --------------------------------------------
+    code = driver_creator.get_driver_as_string(invoke.schedule, [], [],
+                                               "extract", "_post",
+                                               ("region", "name"))
+    assert code == ""
+    out, _ = capsys.readouterr()
+    assert (f"Cannot create driver for 'region-name' because:\nLFRic builtin "
+            f"'{name}' is not supported" in out)
+
+    # write_driver prints an error message, and does not return an error
+    # ------------------------------------------------------------------
+    driver_creator.write_driver(invoke.schedule, [], [],
+                                "extract", "_post", ("region", "name"))
+    assert (f"Cannot create driver for 'region-name' because:\nLFRic builtin "
+            f"'{name}' is not supported" in out)
+
+
+# ----------------------------------------------------------------------------
+def test_lfric_driver_array_of_fields():
+    '''Check that array accesses are created correctly, which is required
+    for builtins. E.g. the following code needs to be created:
+        do df = loop0_start, loop0_stop, 1
+            f2(df) = a + f1(df)
+        enddo
+    As opposed to a normal kernel (where the whole field is passed in),
+    here we need explicit array accesses.
+    '''
+
+    _, invoke = get_invoke("15.1.8_a_plus_X_builtin_array_of_fields.f90",
+                           API, dist_mem=False, idx=0)
+    dep = DependencyTools()
+    input_list, output_list = dep.get_in_out_parameters(invoke.schedule)
+    driver_creator = LFRicExtractDriverCreator()
+
+    driver = driver_creator.\
+        get_driver_as_string(invoke.schedule, input_list, output_list,
+                             "extract", "_post",
+                             region_name=("region", "name"))
+
```

Comment:
Perhaps check that `f1` is read from file too?

Reply:
Done. I added a few more lines to check that the important statements are all there.